ward/testing.py

```diff
@@ -472,6 +472,10 @@ def display_name(self):
         assert len(display_names) == len(TestOutcome)
         return display_names[self]
 
+    @property
+    def is_bad(self) -> bool:
```

Comment:
@darrenburns any thoughts on what to call "a test outcome that represents a failure but may not literally be `FAIL`"?

Reply:
Some ideas:
* `is_fail_or_xpass` - Explicit, although there is a possibility that there could be an `ERR` test outcome too in the future.
* `causes_session_failure` - We could categorise outcomes in outcomes that cause the test suite to fail, and those that don't.  The occurrence of a single `XPASS` or a `FAIL` in a suite causes the overall test session to end in a failure.